docs/custom-flows/oauth-connections.mdx

```diff
@@ -5,110 +5,75 @@ description: Learn how to leverage the Clerk SDK to build completely custom OAut
-
-<CodeBlockTabs type="framework" options={["React"]}>
-```jsx
-import React from "react";
-import {
-  BrowserRouter,
-  Switch,
-  Route,
-} from "react-router-dom";
-import { OAuthStrategy } from "@clerk/types";
-import {
-  ClerkProvider,
-  ClerkLoaded,
-  AuthenticateWithRedirectCallback,
-  UserButton,
-  useSignIn,
-} from "@clerk/clerk-react";
-
-const publishableKey = process.env.REACT_APP_CLERK_PUBLISHABLE_KEY;
-
-function App() {
-  return (
-    //  react-router-dom requires your app to be wrapped with a Router
-    <BrowserRouter>
-      <ClerkProvider publishableKey={publishableKey}>
-        <Switch>
-          {/* Define a / route that displays the OAuth buttons */}
-          <Route path="/">
-            <SignedOut>
-              <SignInOAuthButtons />
-            </SignedOut>
-            <SignedIn>
-              <UserButton afterSignOutAllUrl="/" />
-            </SignedIn>
-          </Route>
-         
-           {/* Define a /sso-callback route that handle the OAuth redirect flow */}
-          <Route path="/sso-callback">
-            <SSOCallback />
-          </Route>
-        </Switch>
-      </ClerkProvider>
-    </BrowserRouter>
-  );
-}
-
-function SSOCallback() {
-  // Handle the redirect flow by rendering the
-  // prebuilt AuthenticateWithRedirectCallback component.
-  // This is the final step in the custom OAuth flow
-  return <AuthenticateWithRedirectCallback />;
-}
-
-function SignInOAuthButtons() {
-  const { signIn } = useSignIn();
-
-  const signInWith = (strategy: OAuthStrategy) => {
-    return signIn.authenticateWithRedirect({
-      strategy,
-      redirectUrl: "/sso-callback",
-      redirectUrlComplete: "/",
-    });
-  };
-
-  // Render a button for each supported OAuth provider
-  // you want to add to your app
-  return (
-    <div>
-      <button onClick={() => signInWith("oauth_google")}>
-        Sign in with Google
-      </button>
-    </div>
-  );
-}
-
-export default App;
-```
-</CodeBlockTabs>
-
-To initiate an OAuth flow for a user that is already signed in, you can use the [`user.createExternalAccount(params)`](/docs/references/javascript/user/create-metadata#create-external-account) method, where `user` is a reference to the currently signed in user.
-
-```jsx
-user
-  .createExternalAccount({ strategy: strategy, redirectUrl: 'your-redirect-url' })
-  .then(externalAccount => {
-    // navigate to
-    // externalAccount.verification!.externalVerificationRedirectURL
-  })
-  .catch(err => {
-    // handle error
-  });
-```
+2. Create a route at the URL that the `redirectUrl` param points to. It's recommended to name this route `/sso-callback`. This route should call the [`Clerk.handleRedirectCallback()`](/docs/references/javascript/clerk/handle-navigation#handle-redirect-callback) method or simply render the prebuilt [`<AuthenticateWithRedirectCallback/>`](/docs/components/control/authenticate-with-callback) component.
```

Comment:
Why do we recommend naming it `/sso-callback`? Are there any technical reasons, or is this just the most common name for this kind of route?

Reply:
@royanger do you have an answer here?

Reply:
it's just a suggestion- not a hard recommendation. I updated the copy 👍